packages/sdks/javascript/src/lib/Ninetailed.spec.ts

```diff
@@ -24,7 +24,10 @@ const mockProfile = (plugins: NinetailedPlugin[] = []) => {
     location: {},
     session: {},
   });
-  const ninetailed = new Ninetailed(apiClient, { plugins });
+  const ninetailed = new Ninetailed(apiClient, {
+    plugins,
+    componentViewTrackingThreshold: 0,
```

Comment:
@maroun9t I had to set the `componentViewTrackingThreshold` to 0 as the mock intersectionObserver always fires immediately. How to make it fire with delays?

Reply:
We use fake timers in the unit test, so the delays don't actually happen. `jest.runAllTimers();` simulates that `setTimeout` has fired.  Check any unit test under `'Observing elements and triggering component views'`